test/wallet-test.js

```diff
@@ -2051,6 +2065,190 @@ describe('Wallet', function() {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(transfer, block);
+      await recip.txdb.add(transfer, block);
+
+      // Check
+      senderBal = await wallet.getBalance();
+      assert.strictEqual(senderBal.tx, 6);
+      assert.strictEqual(senderBal.coin, 4);
+      assert.strictEqual(senderBal.confirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.unconfirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.ulocked, secondHighest);
+      assert.strictEqual(senderBal.clocked, secondHighest);
+
+      recipBal = await recip.getBalance();
+      assert.strictEqual(recipBal.tx, 0);
+      assert.strictEqual(recipBal.coin, 0);
+      assert.strictEqual(recipBal.confirmed, 0);
+      assert.strictEqual(recipBal.unconfirmed, 0);
+      assert.strictEqual(recipBal.ulocked, 0);
+      assert.strictEqual(recipBal.clocked, 0);
+    });
+
+    it('should send and confirm FINALIZE', async () => {
+      // Advance past lockup period
+      wdb.height += network.names.transferLockup;
+
+      const finalize = await wallet.sendFinalize(name, {hardFee: fee});
+
+      // Check
+      let senderBal = await wallet.getBalance();
+      assert.strictEqual(senderBal.tx, 7);
+      // One less wallet coin because name UTXO belongs to recip now
+      assert.strictEqual(senderBal.coin, 3);
+      assert.strictEqual(senderBal.confirmed, 10e6 - (5 * fee));
+      assert.strictEqual(senderBal.unconfirmed, 10e6 - secondHighest - (6 * fee));
+      assert.strictEqual(senderBal.ulocked, 0);
+      assert.strictEqual(senderBal.clocked, secondHighest);
+
+      // The name and its locked value now belong to recipient
+      let recipBal = await recip.getBalance();
+      assert.strictEqual(recipBal.tx, 1);
+      assert.strictEqual(recipBal.coin, 1);
+      assert.strictEqual(recipBal.confirmed, 0);
+      assert.strictEqual(recipBal.unconfirmed, secondHighest);
+      assert.strictEqual(recipBal.ulocked, secondHighest);
+      assert.strictEqual(recipBal.clocked, 0);
+
+      // Confirm FINALIZE
+      const block = {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(finalize, block);
+      await recip.txdb.add(finalize, block);
+      finalizeBlock = block.height;
+
+      // Check
+      senderBal = await wallet.getBalance();
+      assert.strictEqual(senderBal.tx, 7);
+      assert.strictEqual(senderBal.coin, 3);
+      assert.strictEqual(senderBal.confirmed, 10e6 - secondHighest - (6 * fee));
+      assert.strictEqual(senderBal.unconfirmed, 10e6 - secondHighest - (6 * fee));
+      assert.strictEqual(senderBal.ulocked, 0);
+      assert.strictEqual(senderBal.clocked, 0);
+
+      recipBal = await recip.getBalance();
+      assert.strictEqual(recipBal.tx, 1);
+      assert.strictEqual(recipBal.coin, 1);
+      assert.strictEqual(recipBal.confirmed, secondHighest);
```

Comment:
Shouldn't the unlocked values be at 0? The value in the `FINALIZE` is locked and confirmed at this point. Doesn't having `confirmed` and `unconfirmed` equal to `secondHighest` imply that there is value that can be sent when there isn't in this case?